src/controllers/handlers/deployEntityHandler.ts

```diff
@@ -75,9 +74,9 @@ export async function deployEntity(
     }
 
     // validate that the signer has permissions to deploy this scene
-    const addressesWithPermission = await fetchAllowedAddresses(ctx.components)
-    if (!addressesWithPermission.some(($) => $.toLowerCase() == signer.toLowerCase())) {
-      return Error400("Deployment failed: Your wallet has no permission to publish to this server.")
+    const hasPermission = await checkPermissionForAddress(ctx.components, signer)
+    if (!hasPermission) {
+      return Error400(`Deployment failed: Your wallet has no permission to publish to this server.`)
```

Comment:
```suggestion
      return Error400(`Deployment failed: Your wallet has no permission to publish to this server because it doesn't own a Decentraland NAME.`)
```